src/main/webapp/app/shared/form/filepattern-validator.directive.ts

```diff
@@ -0,0 +1,60 @@
+import { AbstractControl, ValidatorFn } from '@angular/forms';
+
+export function filePatternValidator(): ValidatorFn {
+    return (control: AbstractControl): { [key: string]: any } | null => {
+        return validateFilePattern(control.value) ? null : { forbidden: { value: control.value } };
+    };
+}
+
+function validateFilePattern(input: String | null): boolean {
+    let allowed: boolean;
+    let invalidPatterns = 0;
+    const validPatterns: String[] = [
```

Comment:
I don't think it is a good idea to check all file patterns again on the server side after we already had some validation in the client. Users will be really confused, if for some reason these pattern lists are out of sync reason and they suddenly get an error after clicking on save. 

As an example, you could either just fetch this list from the server when opening the page, or maybe send these values after the user opened Artemis using our `/management/info` call and then store them in some file upload exercise config.